web/src/routes/(user)/share/[key]/[[photos=photos]]/[[assetId=id]]/+page.ts

```diff
@@ -2,19 +2,23 @@ import { getAssetThumbnailUrl, setSharedLink } from '$lib/utils';
 import { authenticate } from '$lib/utils/auth';
 import { getFormatter } from '$lib/utils/i18n';
 import { getAssetInfoFromParam } from '$lib/utils/navigation';
-import { getMySharedLink, isHttpError } from '@immich/sdk';
+import { getMySharedLink, isHttpError, getConfig } from '@immich/sdk';
 import type { PageLoad } from './$types';
 
 export const load = (async ({ params }) => {
   const { key } = params;
   await authenticate({ public: true });
   const asset = await getAssetInfoFromParam(params);
+  const config = await getConfig()
+  const domain = config.server.externalDomain;
```

Comment:
You can move these two in the try/catch and put `getAssetInfoFromParam`, `getConfig` and `getMySharedLink` in a `Promise.all` call. It's better to do this in parallel instead of waiting for each to finish.